src/main/java/frc/robot/subsystems/VisionSystem.java

```diff
@@ -0,0 +1,110 @@
+package frc.robot.subsystems;
+
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import edu.wpi.first.wpilibj.shuffleboard.BuiltInLayouts;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardLayout;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.LimelightHelpers;
+import frc.robot.Constants.VisionConstants;
+
+public class VisionSystem extends SubsystemBase {
+    private final double limelightMountAngleRadiansY = VisionConstants.limelightMountAngleRadiansY;
+    private final double limelightMountAngleRadiansX = VisionConstants.limelightMountAngleRadiansX;
+
+    private final double limelightLensHeightMeters = VisionConstants.limelightLensHeightMeters;
+    private final double aprilTagHeightMeters = VisionConstants.aprilTagHeightMeters;
+
+    private NetworkTable limelightTable = NetworkTableInstance.getDefault().getTable(VisionConstants.limelightName);
+    private NetworkTableEntry tableX = limelightTable.getEntry("tx");
+    private NetworkTableEntry tableY = limelightTable.getEntry("ty");
+    private NetworkTableEntry tableArea = limelightTable.getEntry("ta");
+
+    private final double EPSILON = 0.0000001;
+
+    public VisionSystem() {
+        displayToShuffleBoard();
+    }
+
+    private void displayToShuffleBoard() {
+        ShuffleboardLayout visionLayout = Shuffleboard.getTab("Vision")
+                .getLayout("April Tags", BuiltInLayouts.kList);
+        visionLayout.addDouble("X Displacement", () -> getXRadians());
+        visionLayout.addDouble("Y Displacement", () -> getYRadians());
+        visionLayout.addDouble("Area", () -> getArea());
+        visionLayout.addBoolean("Is Detecting", () -> isDetected());
+        visionLayout.addDouble("Distance Meters X", () -> getDistanceMetersX());
+        visionLayout.addDouble("Distance Meters Y", () -> getDistanceMetersY());
+
+        visionLayout.addDouble("X tangent", () -> Math.tan(getXRadians()));
+        visionLayout.addDouble("Y tangent", () -> Math.tan(getYRadians()));
+    }
+
+    public double getXPose() {
+        return LimelightHelpers.getBotPose2d(VisionConstants.limelightName).getX();
```

Comment:
Is X,Y for a particular April tag?  As I mentioned in another comment:
a) What happens if multiple are in-view, as they are when we approach the speaker?  (it has 2)
b) Do we specify the specific April tag *ID* we want to lock-in on?